### DISCUSSION TITLE
View rendering issue in flutter with android webRTC

### DISCUSSION THREAD

**Discussion by jaykumarvcx** [flutter-webrtc/flutter-webrtc#1233] [General]
Hi All,

I need quick help, 

I have developed a wrapper in flutter using android webRTC library to maintain the single code based for flutter and android.
In android webRTC used the  **SurfaceViewRander** to play local Stream as well as remote stream.

Everything was working as expected with flutter 2.5.3 and below but when I updated my flutter sdk with the latest one, my native view(**SurfaceViewRander**) is behaving very strangely.

It is always overlapping on top of  all the view hierarchy and none of the flutter view are showing. Even though I am navigating to the next page, **SurfaceViewRander** was on top of that. I have applied many property on **SurfaceViewRander** like

setZOrderMediaOverlay == true/false
setZOrderOnTop = true/false
setTranslationZ
setZ
But none of these properties are working.

The actual issue is when I am adding my **SurfaceViewRander** to the flutter view it is ok all other views in flutter are visible, but as soon as video come from media channel and load in **SurfaceViewRander**, My all other view goes behind this.

I have invest more than 5-6days in that applying all property of **SurfaceViewRander** and **SurfaceView** bit nothing is working.

I will very much appreciate for quick help.

**Comment by jaykumarvcx** [comment]
Hi Guys,

Any lead please.

**Reply by cloudwebrtc** [reply]
I'm not sure, this seems to be a futter texture rendering bug? 
texture is a flutter widgets, it should be rendered in hierarchical order

**Reply by jaykumarvcx** [reply]
Yes but this is happening. while debugging, I felt my all flutter native view is behind of **SurfaceViewRander** (this is the webRTC android class).

is that any way, I can bring my flutter view component as per the request.

Every things was working happily below 2.5.0, and after that, this is broken, currently we have 3.3.0 and no fixes for this.

**Comment by jaykumarvcx** [comment]
Hi Guys,

Any lead please.

**Comment by jaykumarvcx** [comment]
Hi Guys,

I am still fighting with my issues,
Do let me know if any lead from any of the friends.
